cpp/src/reshape/explode.cu

```diff
@@ -67,39 +68,41 @@ std::unique_ptr<table> explode_functor::operator()<list_view>(
    explode column. This unrolls the top level of lists. Then we need to insert the explode column
    back into the table and return it. */
   lists_column_view lc{input_table.column(explode_column_idx)};
-  rmm::device_uvector<size_type> gather_map_indices(lc.child().size(), stream, mr);
-  auto offsets = lc.offsets();
+  auto sliced_child = lc.get_sliced_child(stream);
+  rmm::device_uvector<size_type> gather_map_indices(sliced_child.size(), stream, mr);
 
-  auto offsets_minus_one = thrust::make_transform_iterator(offsets.begin<size_type>(),
-                                                           [] __device__(auto i) { return i - 1; });
-  auto counting_iter     = thrust::make_counting_iterator(0);
+  // sliced columns can make this a little tricky. We have to start iterating at the start of the
+  // offsets for this column, which could be > 0. Then we also have to handle rebasing the offsets
+  // as we go.
+  auto offsets           = lc.offsets().begin<size_type>() + lc.offset();
+  auto offsets_minus_one = thrust::make_transform_iterator(
+    offsets, [offsets] __device__(auto i) { return i - offsets[0] - 1; });
+  auto counting_iter = thrust::make_counting_iterator(0);
 
   thrust::lower_bound(rmm::exec_policy(stream),
-                      offsets_minus_one,
-                      offsets_minus_one + offsets.size(),
+                      offsets_minus_one + 1,
+                      offsets_minus_one + lc.offsets().size(),
```

Comment:
I don't think this is right.   Size of the offsets will not take slicing of the column into account.  So if you had a column with 99 underlying rows, `.offsets().size()` would be 100.  But if you were looking at a column_view that was sliced to only have  1 row in it, that would definitely be wrong.  I think you want `(lc.size()+1)`, since the lists_column_view here will return the sliced size.